tests/unit/graph_learning_test.py

```diff
@@ -1661,18 +1727,36 @@ def test_5lm_feature_experiment(self):
         config = copy.deepcopy(self.feature_5lm_config)
         with MontyObjectRecognitionExperiment(config) as exp:
             pprint("...training...")
-            exp.train()
-
-            train_stats = pd.read_csv(
-                os.path.join(exp.output_dir, "train_stats.csv")
-            )
-            # The following check is brittle and depends on sensor arrangement. Leaving
-            # the rest of the test intact to detect run failures, but disabling checking
-            # of particular results.
-            # self.check_multilm_train_results(train_stats, num_lms=5, min_done=3)
+            objects = [self.fake_obs_learn, self.fake_obs_house_3d]
+            graph_modules = self.get_5lm_gm_with_fake_objects(objects)
 
             pprint("...evaluating...")
-            exp.evaluate()
+            monty = exp.model
+            monty.set_experiment_mode("eval")
+            monty.learning_modules = [gm.learning_module for gm in graph_modules]
+
+            for gm in graph_modules:
+                gm.mode = "eval"
+                gm.pre_episode(self.placeholder_target)
+
+            exp.pre_epoch()
+            # Use the first graph module's observations for the episodes
+            # since it has no offsets like the others do.
```

Comment:
Why does this matter? I thought we are just using this to know how many episodes there are and how many steps in each episode but the observations themselves are extracted for each lm separately on line 1752

Reply:
Your right, when I was refactoring this code, I think I misread what we were doing. This isn't accurate at all.